### ISSUE TITLE
Repositories with (big?) editorconfig can't load

### ISSUE THREAD

**Issue by markusamshove** [go-gitea/gitea#29402]
### Description

Since the update to 1.21.6 I can't view some "home" pages of repositories, resulting in the following logs:

```
2024/02/25 19:13:22 ...eb/routing/logger.go:68:func1() [W] router: slow      GET /org/repo for ....:0, elapsed 3280.9ms @ repo/middlewares.go:17(repo.SetEditorconfigIfExists)
2024/02/25 19:14:18 ...ers/web/repo/view.go:675:checkHomeCodeViewable() [E] ReadBy: context canceled
2024/02/25 19:14:18 .../context_response.go:68:HTML() [D] Template: status/500
2024/02/25 19:14:18 ...dules/git/command.go:344:Run() [D] slow git.Command.Run: /usr/bin/git -c protocol.version=2 -c credential.helper= -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= cat-file --batch-check (1m0.028896423s)
2024/02/25 19:14:18 ...dules/git/command.go:344:Run() [D] slow git.Command.Run: /usr/bin/git -c protocol.version=2 -c credential.helper= -c filter.lfs.required= -c filter.lfs.smudge= -c filter.lfs.clean= cat-file --batch (1m0.029096177s)
2024/02/25 19:14:18 ...eb/routing/logger.go:102:func1() [I] router: completed GET /org/repo for ....:0, 500 Internal Server Error in 60041.7ms @ repo/view.go:730(repo.Home)
2024/02/25 19:14:18 .../context_response.go:68:HTML() [D] Template: status/404
2024/02/25 19:14:18 ...eb/routing/logger.go:102:func1() [I] router: completed GET /error/HTTP_BAD_GATEWAY.html.var for 10.120.28.76:0, 404 Not Found in 8.9ms @ context/repo.go:425(context.RepoAssignment)
```

I've cloned one repository that doesn't work in our instance to try.gitea, but there it works fine:
https://try.gitea.io/MarkusAmshove/natls

If I open `/org/repo/pulls` in the browser I instantly get the pulls site. I can even go into pull requests, but once I click on the `Files changed` tab it hangs again.
I guess that it has something to do with the editorconfig, since its logged as warning and both the `Files changed` tab and home view of the repository might use it (for the README?).

In the example repository on try.gitea you can see that the editorconfig is quite big. Some other repositories that have rather small editorconfigs work fine.

Reverting to 1.21.5 resolved the issue.

### Gitea Version

1.21.6

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

Downloaded amd64 linux binary, using systemd.

### Database

PostgreSQL

**Comment by delvh** [comment]
Hmm… according to the logs, my primary suspicion is https://github.com/go-gitea/gitea/pull/29310.
However, I have no idea how that might be related.
Could it be that your Gitea instance is run on Windows?

**Comment by MarkusAmshove** [comment]
It's SLES 15, run with the amd64 ELF binary, no docker

**Comment by derelm** [comment]
Thanks @MarkusAmshove i am experiencing something similar; for me the page just never finishes (or maybe i have just never been patient enough to wait for more than a couple of minutes).
The logs are roughly the same - no real error message but just a warning that the router is slow

**Comment by RouxAntoine** [comment]
same for me on repository home page: gitea version v1.21.6

```
2024/02/25 23:57:36 ...eb/routing/logger.go:102:func1() [I] router: completed GET /kotlin/nextcloud-uploader for 192.168.2.254:0, 500 Internal Server Error in 228.1ms @ repo/view.go:730(repo.Home)
2024/02/25 23:57:36 ...ers/web/repo/view.go:675:checkHomeCodeViewable() [E] ReadBy: context canceled
```

**Comment by achikhv** [comment]
Same problem. editorconfig is 144 lines and 6.8kB if it is relevant.

**Comment by wxiaoguang** [comment]
Please use Gitea 1.21 nightly, it should have been fixed there.

**Comment by lunny** [comment]
Fixed by #29370 and will be released on v1.21.7
